flasher.cpp

```diff
@@ -440,24 +448,18 @@ bool Flasher::CollectBoardId()
 {
     bool success = false;
 
-    if (serial_port_.isOpen()) {
-        serial_port_.write(kBoardIdCmd, sizeof(kBoardIdCmd));
-        serial_port_.waitForReadyRead(kCollectBoardIdSerialTimeoutInMs);
-        QByteArray data = serial_port_.readAll();
-        QByteArray board_id = data.left(kBoardIdSize);
-        QByteArray data_crc = data.right(kCrc32Size);
-
-        uint32_t crc = Deserialize32((uint8_t*)data_crc.data());
-        uint32_t calc_crc = crc::CalculateCrc32((uint8_t*)board_id.data(), static_cast<uint32_t>(kBoardIdSize), false, false);
+    QByteArray out_data;
+    if(ReadMessageWithCrc(kBoardIdCmd, sizeof(kBoardIdCmd), kCollectBoardIdSerialTimeoutInMs, out_data)) {
 
-        if ((board_id.size() == kBoardIdSize) && (calc_crc == crc)) {
-            board_id_ = board_id.toHex();
+        if(out_data.size() == kBoardIdSize) {
+            board_id_ = out_data.toBase64();
             qInfo() << "Board ID: " << board_id_;
             success = true;
         }
-        else {
-            qInfo() << "Board id error";
-        }
+    }
+
+    if(!success) {
+        qInfo() << "Board id error";
     }
```

Comment:
I am not sure that we need this debug message. In case of fail, we will ShowTextInBrowser("Board ID error....").

Reply:
For console?

Reply:
Maybe it will be better to put this qInfo in main. But we can do it in the future.